src/utils/router_state.rs

```diff
@@ -260,15 +262,116 @@ pub fn is_vertiport_available(
+    existing_flight_plans
+        .iter()
+        .filter(|flight_plan| {
+            flight_plan
+                .data
+                .as_ref()
+                .unwrap()
+                .destination_vertiport_id
+                .as_ref()
+                .unwrap()
+                == vertiport_id
+                && flight_plan
+                    .data
+                    .as_ref()
+                    .unwrap()
+                    .scheduled_arrival
+                    .as_ref()
+                    .unwrap()
+                    .seconds // arrival time needs to be less than 2x time needed - to allow landing and and then take off again)
+                    < timestamp.timestamp() + LANDING_AND_UNLOADING_TIME_MIN as i64 * 60
+        })
+        .for_each(|flight_plan| {
+            if vehicles_plans_sorted.contains_key(&flight_plan.data.as_ref().unwrap().vehicle_id) {
+                vehicles_plans_sorted
+                    .get_mut(&flight_plan.data.as_ref().unwrap().vehicle_id)
+                    .unwrap()
+                    .push(flight_plan.clone());
+            } else {
+                vehicles_plans_sorted.insert(
+                    flight_plan.data.as_ref().unwrap().vehicle_id.clone(),
+                    vec![flight_plan.clone()],
+                );
+            }
+        });
+    //sort by scheduled arrival, latest first
+    vehicles_plans_sorted
+        .iter_mut()
+        .for_each(|(_, flight_plans)| {
+            flight_plans.sort_by(|a, b| {
+                b.data
+                    .as_ref()
+                    .unwrap()
+                    .scheduled_arrival
+                    .as_ref()
+                    .unwrap()
+                    .seconds
+                    .cmp(
+                        &a.data
+                            .as_ref()
+                            .unwrap()
+                            .scheduled_arrival
+                            .as_ref()
+                            .unwrap()
+                            .seconds,
+                    )
+            });
+        });
+    //return only the latest flight plan for each vehicle
+    let vehicles = vehicles_plans_sorted
+        .iter()
+        .map(|(vehicle_id, flight_plans)| {
+            let mut minutes_to_arrival = (flight_plans
+                .first()
+                .unwrap()
+                .data
+                .as_ref()
+                .unwrap()
+                .scheduled_arrival
+                .as_ref()
+                .unwrap()
+                .seconds
+                - timestamp.timestamp())
+                / 60;
+            if minutes_to_arrival < 0 {
+                minutes_to_arrival = 0;
+            }
+            (vehicle_id.clone(), minutes_to_arrival)
+        })
+        .collect();
+    debug!(
+        "Vehicles at vertiport: {} at a time: {} : {:?}",
+        &vertiport_id, timestamp, vehicles
```

Comment:
If we only want the latest flight for each vehicle, wouldn't it be more efficient to use a priority queue of size 1 as the value type of `vehicles_plans_sorted`. 

Populating the hashmap takes O(n*m) for m is the average length of the flight plans and n is the number of vehicles, and sorting would be O(n * (mlogm)), then the collection will be O(n), so total runtime is O(n*(m + n + mlogm)).

If priority queue is used, we can achieve O(n*mlogm) for populating the hashmap, then we skip the sorting, and go straight to collecting. So, overall runtime would be O(n * (mlogm + n)). We can improve the performance by a factor of n*m.

Reply:
You are probably right :) , but it would take more time, which I don't have now. Might be a good optimisation for R3